# Data Loading
Tujuan: Melihat dan memuat data.

## 1. Bentuk dan Ukuran Data

Pertama, kita perlu melihat seberapa banyak data yang ada.

In [6]:
import pandas as pd

In [7]:
results = pd.read_csv("../data/raw/results.csv")
former_names = pd.read_csv("../data/raw/former_names.csv")
goalscorers = pd.read_csv("../data/raw/goalscorers.csv")
shootouts = pd.read_csv("../data/raw/shootouts.csv")

In [10]:
data_data = [results, former_names, goalscorers, shootouts]

for data in data_data:
    print(data.shape)

(49477, 9)
(36, 4)
(47690, 8)
(678, 5)


### Insight:
- `results` : Memiliki 49477 baris data dengan 9 kolom.
- `former_names` : Memiliki 36 baris data dengan 4 kolom.
- `goalscorers`: Memiliki 47690 baris data dengan 8 kolom.
- `shootouts`: Memiliki 678 baris data dengan 5 kolom.

## 2. Definisi Setiap Data

Kita perlu memahami maksud dan definisi dari setiap data dan kolom.

In [13]:
results.head()

,date,home_team,away_team,home_score,away_score,tournament,city,country,neutral
0,1872-11-30,Scotland,England,0.0,0.0,Friendly,Glasgow,Scotland,False
1,1873-03-08,England,Scotland,4.0,2.0,Friendly,London,England,False
2,1874-03-07,Scotland,England,2.0,1.0,Friendly,Glasgow,Scotland,False
3,1875-03-06,England,Scotland,2.0,2.0,Friendly,London,England,False
4,1876-03-04,Scotland,England,3.0,0.0,Friendly,Glasgow,Scotland,False


### Insight
Tabel `results`

Tabel ini menyimpan data historis hasil pertandingan internasional. Data ini terdiri dari 9 kolom yang merangkum rincian setiap laga:

*   **`date`**: Tanggal kapan pertandingan tersebut dilaksanakan (dengan format Tahun-Bulan-Tanggal).
*   **`home_team`**: Nama negara atau tim yang bertindak sebagai tuan rumah (bermain di kandang).
*   **`away_team`**: Nama negara atau tim yang bertindak sebagai tamu (lawan).
*   **`home_score`**: Skor atau jumlah gol yang berhasil dicetak oleh tim tuan rumah.
*   **`away_score`**: Skor atau jumlah gol yang berhasil dicetak oleh tim tamu.
*   **`tournament`**: Jenis kompetisi atau ajang pertandingan tersebut digelar (misalnya *Friendly* yang berarti laga persahabatan).
*   **`city`**: Nama kota tempat pertandingan diselenggarakan.
*   **`country`**: Nama negara tempat pertandingan diselenggarakan.
*   **`neutral`**: Keterangan (*True/False*) yang menunjukkan apakah pertandingan dimainkan di tempat netral. Jika bernilai `False`, berarti pertandingan benar-benar dilakukan di kandang tim tuan rumah, bukan di lokasi pihak ketiga.

In [14]:
former_names.head()

,current,former,start_date,end_date
0,Benin,Dahomey,1959-11-08,1975-11-30
1,Burkina Faso,Upper Volta,1960-04-14,1984-08-04
2,Curaçao,Netherlands Antilles,1957-03-03,2010-10-10
3,Czechoslovakia,Bohemia,1903-04-05,1919-01-01
4,Czechoslovakia,Bohemia and Moravia,1939-01-01,1945-05-01


### Insight
Tabel `former_names`

Tabel ini berfungsi sebagai kamus sejarah yang melacak perubahan nama suatu negara atau tim nasional dari waktu ke waktu. Tabel ini sangat berguna untuk memastikan data pertandingan masa lalu (dengan nama lama) tetap bisa dihubungkan dengan negara modernnya. Data ini terdiri dari 4 kolom:

*   **`current`**: Nama resmi negara atau tim tersebut saat ini.
*   **`former`**: Nama lama, julukan historis, atau entitas politik terdahulu dari negara tersebut.
*   **`start_date`**: Tanggal kapan nama lama tersebut mulai digunakan atau tercatat secara resmi.
*   **`end_date`**: Tanggal kapan penggunaan nama lama tersebut berakhir (kapan entitas tersebut berganti nama atau bubar).



In [15]:
goalscorers.head()

,date,home_team,away_team,team,scorer,minute,own_goal,penalty
0,1916-07-02,Chile,Uruguay,Uruguay,José Piendibene,44.0,False,False
1,1916-07-02,Chile,Uruguay,Uruguay,Isabelino Gradín,55.0,False,False
2,1916-07-02,Chile,Uruguay,Uruguay,Isabelino Gradín,70.0,False,False
3,1916-07-02,Chile,Uruguay,Uruguay,José Piendibene,75.0,False,False
4,1916-07-06,Argentina,Chile,Argentina,Alberto Ohaco,2.0,False,False


### Insight
Tabel `goalscorers`

Tabel ini menyimpan catatan detail dari setiap gol yang tercipta di berbagai pertandingan. Ini adalah data tingkat pemain (bukan sekadar hasil akhir tim). Data ini terdiri dari 8 kolom:

*   **`date`**: Tanggal kapan pertandingan tersebut berlangsung.
*   **`home_team`**: Nama tim tuan rumah.
*   **`away_team`**: Nama tim tamu.
*   **`team`**: Nama tim dari pemain yang mencetak gol (berguna untuk mencocokkan apakah gol dicetak oleh tim tuan rumah atau tamu).
*   **`scorer`**: Nama lengkap atau panggil dari pemain yang berhasil mencetak gol.
*   **`minute`**: Menit pertandingan ke berapa gol tersebut bersarang di gawang.
*   **`own_goal`**: Indikator (*True/False*) apakah gol tersebut merupakan gol bunuh diri. Jika `True`, berarti pemain secara tidak sengaja memasukkan bola ke gawangnya sendiri.
*   **`penalty`**: Indikator (*True/False*) apakah gol tersebut dicetak melalui titik putih (tendangan penalti).


In [16]:
shootouts.head()

,date,home_team,away_team,winner,first_shooter
0,1967-08-22,India,Taiwan,Taiwan,NaN
1,1971-11-14,South Korea,Vietnam Republic,South Korea,NaN
2,1972-05-07,South Korea,Iraq,Iraq,NaN
3,1972-05-17,Thailand,South Korea,South Korea,NaN
4,1972-05-19,Thailand,Cambodia,Thailand,NaN


### Insight
Tabel `shootouts`

Tabel ini secara khusus merangkum hasil akhir dari pertandingan-pertandingan yang berakhir imbang dan harus ditentukan melalui drama adu penalti. Data ini terdiri dari 5 kolom:

*   **`date`**: Tanggal terjadinya pertandingan adu penalti tersebut.
*   **`home_team`**: Nama tim yang bertindak sebagai tuan rumah.
*   **`away_team`**: Nama tim lawan (tamu).
*   **`winner`**: Nama tim yang pada akhirnya memenangkan babak adu penalti.
*   **`first_shooter`**: Nama tim yang mendapatkan giliran pertama untuk menendang bola dalam adu penalti. (Catatan: Nilai `NaN` yang terlihat pada baris awal menunjukkan bahwa informasi ini tidak tercatat atau kosong pada pertandingan tersebut).